pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer_test.go

```diff
@@ -24,21 +24,67 @@ func TestApplyLocalitySetting(t *testing.T) {
 	}
 
 	t.Run("Distribute", func(t *testing.T) {
-		g := NewGomegaWithT(t)
-		env := buildEnvForClustersWithDistribute()
-		cluster := buildFakeCluster()
-		ApplyLocalityLBSetting(locality, cluster.LoadAssignment, env.Mesh.LocalityLbSetting)
-
-		for _, localityEndpoint := range cluster.LoadAssignment.Endpoints {
-			if util.LocalityMatch(localityEndpoint.Locality, "region1/zone1/subzone1") {
-				g.Expect(localityEndpoint.LoadBalancingWeight.GetValue()).To(Equal(uint32(90)))
-				continue
-			}
-			if util.LocalityMatch(localityEndpoint.Locality, "region1/zone1") {
-				g.Expect(localityEndpoint.LoadBalancingWeight.GetValue()).To(Equal(uint32(5)))
-				continue
-			}
-			g.Expect(localityEndpoint.LbEndpoints).To(BeNil())
+		tests := []struct {
+			name       string
+			distribute []*meshconfig.LocalityLoadBalancerSetting_Distribute
+			expected   []int
+		}{
+			{
+				name: "distribution between subzones",
+				distribute: []*meshconfig.LocalityLoadBalancerSetting_Distribute{
+					{
+						From: "region1/zone1/subzone1",
+						To: map[string]uint32{
+							"region1/zone1/subzone1": 80,
+							"region1/zone1/subzone2": 15,
+							"region1/zone1/subzone3": 5,
+						},
+					},
+				},
+				expected: []int{40, 40, 15, 5, 0, 0, 0},
+			},
+			{
+				name: "distribute with overlapping to",
```

Comment:
This test can not run into validation

Reply:
I don't think this is an invalid configuration. Why should we not support this? 

The logic seems to make sense to me.

If we don't want to support something like this, then I don't understand why we have all the added complexity of total weight etc.

Reply:
You mentioned previously we should consider their 'original weights'.

The weighting is 2 tiered - first, we use the weights to pick a locallity (this code). Then, within the locallity we use the defined weights for each endpoint (from service entry) to pick within that endpoint. This code here never interacts with user defined weights.